nistats/reporting/_get_clusters_table.py

```diff
@@ -0,0 +1,176 @@
+"""
+This module implements plotting functions useful to report analysis results.
+
+Author: Martin Perez-Guevara, Elvis Dohmatob, 2017
+"""
+
+import warnings
+from string import ascii_lowercase
+
+import numpy as np
+import pandas as pd
+import nibabel as nib
+from scipy import ndimage
+from nilearn.image.resampling import coord_transform
+
+
+def _local_max(data, affine, min_distance):
+    """Find all local maxima of the array, separated by at least min_distance.
+    Adapted from https://stackoverflow.com/a/22631583/2589328
+
+    Parameters
+    ----------
+    data : array_like
+        3D array of with masked values for cluster.
+
+    min_distance : `int`
+        Minimum distance between local maxima in ``data``, in terms of mm.
+
+    Returns
+    -------
+    ijk : `numpy.ndarray`
+        (n_foci, 3) array of local maxima indices for cluster.
+
+    vals : `numpy.ndarray`
+        (n_foci,) array of values from data at ijk.
+    """
+    # Initial identification of subpeaks with minimal minimum distance
+    data_max = ndimage.filters.maximum_filter(data, 3)
+    maxima = (data == data_max)
+    data_min = ndimage.filters.minimum_filter(data, 3)
+    diff = ((data_max - data_min) > 0)
+    maxima[diff == 0] = 0
+    
+    labeled, n_subpeaks = ndimage.label(maxima)
+    ijk = np.array(ndimage.center_of_mass(data, labeled,
+                                          range(1, n_subpeaks + 1)))
+    ijk = np.round(ijk).astype(int)
+    
+    vals = np.apply_along_axis(arr=ijk, axis=1, func1d=_get_val,
+                               input_arr=data)
+    
+    # Sort subpeaks in cluster in descending order of stat value
+    order = (-vals).argsort()
+    vals = vals[order]
+    ijk = ijk[order, :]
+    xyz = nib.affines.apply_affine(affine, ijk)  # Convert to xyz in mm
+    
+    # Reduce list of subpeaks based on distance
+    keep_idx = np.ones(xyz.shape[0]).astype(bool)
+    for i in range(xyz.shape[0]):
+        for j in range(i + 1, xyz.shape[0]):
+            if keep_idx[i] == 1:
+                dist = np.linalg.norm(xyz[i, :] - xyz[j, :])
```

Comment:
This loops seems pretty inefficient if xyz.shape[0] is large. there is room for improvement here.

Reply:
Based on limited profiling the speed hit is minimal.
Not rally sure how to optimize this in a meaningful way. 
Looking into it.